common4j/src/main/com/microsoft/identity/common/java/crypto/DefaultCryptoFactory.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.crypto;
+
+import com.microsoft.identity.common.java.exception.ClientException;
+
+import java.security.KeyFactory;
+import java.security.KeyPairGenerator;
+import java.security.Signature;
+
+import javax.crypto.Cipher;
+import javax.crypto.Mac;
+
+import lombok.NonNull;
+
+/**
+ * A Factory class that generates/utilizes platform's default crypto objects.
+ */
+public class DefaultCryptoFactory implements ICryptoFactory {
```

Comment:
Please add UTs for this class

Reply:
+1

Reply:
@iamgusain 

due to the nature of this class, it's going to be tricky to cover it with UT. (i.e. what algorithms are we going to include as part of the test - every single algorithm exists? algorithms that we use in common/MSAL/broker now? or algorithm that is in scope for FIPS only? How are we going to make sure that the test is updated when someone introduces a new algo randomly in the code?)

Because of this.... I think the cleanest, maintainable approach for this scope is to
1.  add a parameterized tests (switching between DefaultCryptoFactory, and the NonFips and FIPS Crypto factory in broker)
3.  The scope is to cover *only* algorithms we plan to support for FIPS. 

The test will later be added in the broker, as a separate PR.

thoughts?

Reply:
Alternatively, I can get rid of this class, and in common4j, fall back to the behavior of "If no ICryptoFactory is not provided, always fall back to the default implementation"

Reply:
(Going to merge this for now to simply my PR tree... but please do tag/ping me when you response, so I can address this later on).